bench-tps/src/log_transaction_service.rs

```diff
@@ -0,0 +1,483 @@
+        }
+        measure_process_blocks.stop();
+        let time_process_blocks_us = measure_process_blocks.as_us();
+        info!("Time to process {num_blocks} blocks: {time_process_blocks_us}us.");
+    }
+
+    fn process_block(
+        block: UiConfirmedBlock,
+        signature_to_tx_info: &mut MapSignatureToTxInfo,
+        slot: u64,
+        tx_log_writer: &mut TransactionLogWriter,
+        block_log_writer: &mut BlockLogWriter,
+    ) {
+        let rewards = block
+            .rewards
+            .as_ref()
+            .expect("Rewards should be part of the block information.");
+        let slot_leader = rewards
+            .iter()
+            .find(|r| r.reward_type == Some(RewardType::Fee))
+            .map_or("".to_string(), |x| x.pubkey.clone());
+
+        let Some(transactions) = &block.transactions else {
+            warn!("Empty block: {slot}");
+            return;
+        };
+
+        let mut num_bench_tps_transactions: usize = 0;
+        let mut total_cu_consumed: u64 = 0;
+        let mut bench_tps_cu_consumed: u64 = 0;
+        for EncodedTransactionWithStatusMeta {
+            transaction, meta, ..
+        } in transactions
+        {
+            let Some(transaction) = transaction.decode() else {
+                continue;
+            };
+            let cu_consumed = meta
+                .as_ref()
+                .map_or(0, |meta| match meta.compute_units_consumed {
+                    OptionSerializer::Some(cu_consumed) => cu_consumed,
+                    _ => 0,
+                });
+            let signature = &transaction.signatures[0];
+
+            total_cu_consumed = total_cu_consumed.saturating_add(cu_consumed);
+            if let Some(TransactionSendInfo {
+                sent_at,
+                compute_unit_price,
+            }) = signature_to_tx_info.remove(signature)
+            {
+                num_bench_tps_transactions = num_bench_tps_transactions.saturating_add(1);
+                bench_tps_cu_consumed = bench_tps_cu_consumed.saturating_add(cu_consumed);
+
+                tx_log_writer.write(
+                    Some(block.blockhash.clone()),
+                    Some(slot_leader.clone()),
+                    signature,
+                    sent_at,
+                    Some(slot),
+                    block.block_time,
+                    meta.as_ref(),
+                    false,
+                    compute_unit_price,
+                );
+            }
+        }
+        block_log_writer.write(
+            block.blockhash.clone(),
+            slot_leader,
+            slot,
+            block.block_time,
+            num_bench_tps_transactions,
+            transactions.len(),
+            bench_tps_cu_consumed,
+            total_cu_consumed,
+        )
+    }
+
+    fn clean_transaction_map(
+        tx_log_writer: &mut TransactionLogWriter,
+        signature_to_tx_info: &mut MapSignatureToTxInfo,
+    ) {
+        let now: DateTime<Utc> = Utc::now();
```

Comment:
here should be the time of the last processed block because it might be that we lag behind during the processing